utils/generate-playbook

```diff
@@ -0,0 +1,171 @@
+                          'priority': rule['priority']}
+        inheritance.append(representation)
+    return inheritance
+
+
+def koji_tag_packages(packages):
+    """ Transform "listPackages" result into Ansible-formatted data. """
+    packages = sorted(packages, key=itemgetter('package_name'))
+    results = defaultdict(list)
+    for package in packages:
+        owner_name = package['owner_name']
+        results[owner_name].append(package['package_name'])
+    return dict(results)
+
+
+def koji_tag_external_repos(repos):
+    """ Transform "getTagExternalRepos" result into Ansible-formatted data. """
+    results = []
+    repos = sorted(repos, key=itemgetter('priority'))
+    for repo in repos:
+        representation = {'repo': repo['external_repo_name'],
+                          'priority': repo['priority']}
+        results.append(representation)
+    return results
+
+
+def koji_tag_tasks(tags):
+    """ Generate the YAML data for all koji tags. """
+    tasks = []
+    for taginfo in tags:
+        # name = 'configure koji tag %s' % taginfo['name']
+        # Do not configure tag ID numbers:
+        del taginfo['id']
+        # Do not configure things if they're "false" or "None":
+        for key in taginfo.copy():
+            if not taginfo[key]:
+                del taginfo[key]
+        if 'external_repos' in taginfo:
+            parsed_repos = koji_tag_external_repos(taginfo['external_repos'])
+            taginfo['external_repos'] = parsed_repos
+        if 'inheritance' in taginfo:
+            parsed_inheritance = koji_tag_inheritance(taginfo['inheritance'])
+            taginfo['inheritance'] = parsed_inheritance
+        if 'packages' in taginfo:
+            parsed_packages = koji_tag_packages(taginfo['packages'])
+            taginfo['packages'] = parsed_packages
+        task = {
+            # 'name': name,
+            'koji_tag': taginfo,
+        }
+        tasks.append(task)
+    return tasks
+
+
+def generate_playbook(tasks):
+    """ Generate a playbook YAML from a set of tasks. """
+    playbook = {'tasks': tasks}
+    text = yaml.dump(playbook, default_flow_style=False)
+    print(text)
+
+
+def parse_args():
+    """ Parse CLI options """
+    parser = argparse.ArgumentParser(description=DESCRIPTION,
+                                     formatter_class=RawTextHelpFormatter)
+    parser.add_argument('--profile', default='koji',
+                        help='koji client profile (defaults to "cbs")')
+    parser.add_argument('--tag-regex',
+                        help='match a subset of tags (by name)')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    session = common_koji.get_session(args.profile)
+
+    tags = session.listTags()
+
+    # Optionally filter our tag list by a regular expression.
+    if args.tag_regex:
+        regex = re.compile(args.tag_regex)
+        tags = [tag for tag in tags if regex.search(tag['name'])]
+
+    print('querying %i tags' % len(tags))
```

Comment:
This message should go to stderr, otherwise it is mixed with the YAML output.

Reply:
Great catch. Thank you. I've made this message go to stderr in my branch.